src/components/geonode.jsx

```diff
@@ -63,10 +64,23 @@ class GeoNodeViewer extends React.Component {
     this.updateMap(this.props);
     this.mode = this.props.mode || 'viewer';
     this.edit = (this.mode === 'composer');
+    if (this.props.layer) {
+      WMSService.getCapabilities(this.props.wmsServer, this._getLayerInfo.bind(this), function() {}, this.props.proxy);
+    }
   }
   componentWillReceiveProps(props) {
     this.updateMap(props);
   }
+  _getLayerInfo(layerInfo, getMapUrl) {
```

Comment:
Moved most of the code to sdk with: https://github.com/boundlessgeo/sdk/pull/364